model/main.py

```diff
@@ -46,14 +46,19 @@ def inference(self, image, vocab:Vocabulary):
 			word = torch.cuda.IntTensor([vocab.word_to_index['<START>']])
 		else:
 			word = torch.tensor(vocab.word_to_index['<START>'])
-		sentence = [word.tolist()]
-		
+		sentence = [word.item()]
+		attention_weights=[]
 		for i in range(self.caption_max_length):
 			alphas, weighted_features = self.attention.forward(image_features, hidden)
 			predictions_t, hidden = self.decoder.forward(weighted_features, word, hidden)
 			word = torch.argmax(predictions_t, dim=-1)
-			sentence.append(word.tolist())
-			
-			if word[0].tolist()==vocab.word_to_index['<END>']:
+			if word[0].item()==vocab.word_to_index['<END>']:
```

Comment:
I see we send the `Vocabulary` on the constructor, so `self.vocab` is available here. Could we just use that one? So we don't send it as a parameter?